scripts/storage-emulator-integration/tests.ts

```diff
@@ -723,6 +723,41 @@ describe("Storage emulator", () => {
           expect(metadata2).to.deep.equal(metadata1);
         });
 
+        it("should set null custom metadata values to empty strings", async () => {
+          const [, source] = await testBucket.upload(smallFilePath);
+
+          const file = testBucket.file(copyDestinationFile);
+          const metadata = { foo: "bar", nullMetadata: null };
+          const cacheControl = "private,max-age=10,immutable";
+          // Types for CopyOptions are wrong (@google-cloud/storage sub-dependency needs
+          // update to include https://github.com/googleapis/nodejs-storage/pull/1406
+          // and https://github.com/googleapis/nodejs-storage/pull/1426)
+          const copyOpts: CopyOptions & { [key: string]: unknown } = {
+            metadata,
+            cacheControl,
+          };
+          const [, { resource: metadata1 }] = await testBucket
+            .file(smallFilePath.split("/").slice(-1)[0])
+            .copy(file, copyOpts);
+
+          expect(metadata1).to.deep.include({
+            bucket: source.bucket,
+            contentType: source.contentType,
+            etag: source.etag,
+            crc32c: source.crc32c,
+            metadata: {
+              foo: "bar",
+              // Sets null metadata values to empty strings
+              nullMetadata: "",
+            },
+            cacheControl,
+          });
+
+          // Also double check with a new metadata fetch
+          const [metadata2] = await file.getMetadata();
+          expect(metadata2).to.deep.equal(metadata1);
+        });
```

Comment:
This seems weird behaviour but can be confirmed with:

```
curl -i 'https://storage.googleapis.com/storage/v1/b/bookcreator-dev-firetool-tests/o/prefix/rewriteTo/b/bookcreator-dev-firetool-tests/o/copied' -H "authorization: Bearer $(gcloud auth print-access-token)" -H 'content-type: application/json' --data-raw '{ "metadata": { "hello": "world", "foo": null  } }' 

HTTP/2 200 
x-guploader-uploadid: ADPycdtAGoz55T_cY81t5UnGT_MZlTnOeXiBSkCyL6ydj5-1yIPxz4zkg8dPzdIkjZww3q5eKzJN7n_nnVpii6K-iw
content-type: application/json; charset=UTF-8
date: Wed, 11 Aug 2021 08:13:45 GMT
vary: Origin
vary: X-Origin
cache-control: no-cache, no-store, max-age=0, must-revalidate
expires: Mon, 01 Jan 1990 00:00:00 GMT
pragma: no-cache
content-length: 974
server: UploadServer
alt-svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000,h3-T051=":443"; ma=2592000,h3-Q050=":443"; ma=2592000,h3-Q046=":443"; ma=2592000,h3-Q043=":443"; ma=2592000,quic=":443"; ma=2592000; v="46,43"

{
  "kind": "storage#rewriteResponse",
  "totalBytesRewritten": "5",
  "objectSize": "5",
  "done": true,
  "resource": {
    "kind": "storage#object",
    "id": "bookcreator-dev-firetool-tests/copied/1628669625765894",
    "selfLink": "https://www.googleapis.com/storage/v1/b/bookcreator-dev-firetool-tests/o/copied",
    "mediaLink": "https://storage.googleapis.com/download/storage/v1/b/bookcreator-dev-firetool-tests/o/copied?generation=1628669625765894&alt=media",
    "name": "copied",
    "bucket": "bookcreator-dev-firetool-tests",
    "generation": "1628669625765894",
    "metageneration": "1",
    "storageClass": "STANDARD",
    "size": "5",
    "md5Hash": "Z0RBlgyhui3gitTlDJ/emA==",
    "crc32c": "t0Xs5w==",
    "etag": "CIaggJTDqPICEAE=",
    "timeCreated": "2021-08-11T08:13:45.767Z",
    "updated": "2021-08-11T08:13:45.767Z",
    "timeStorageClassUpdated": "2021-08-11T08:13:45.767Z",
    "metadata": {
      "hello": "world",
      "foo": ""
    }
  }
}
```

(Confirmed same behaviour with `copyTo`.)